backend/fms_core/template_importer/row_handlers/sample_submission/sample.py

```diff
@@ -97,13 +100,26 @@ def process_row_inner(self, sample, container, project, parent_container, indivi
             self.errors['sample_kind'] = [f"Sample Kind {sample['sample_kind']} not found"]
         #TODO: sample kind str normalization
 
-
-        sample_obj, self.errors['sample'], self.warnings['sample'] = \
-            create_full_sample(name=sample['name'], volume=sample['volume'], collection_site=sample['collection_site'],
-                               creation_date=sample['creation_date'], coordinates=sample['coordinates'], alias=sample['alias'],
-                               concentration=sample['concentration'], tissue_source=sample['tissue_source'],
-                               experimental_group=sample['experimental_group'], container=container_obj, individual=individual_obj,
-                               sample_kind=sample_kind_obj, comment=comment)
+        # Library are submitted
+        if any([library['library_type'], library['index'], library['platform'], library['strandedness']]):
+            # Create library objects
+            library_type_obj, self.errors['library_type'], self.warnings['library_type']  = get_library_type(library['library_type'])
+            index_obj, self.errors['index'], self.warnings['index']  = get_index(library['index'])
+            platform_obj, self.errors['platform'], self.warnings['platform']  = get_platform(library['platform'])
+
+            library_obj, self.errors['library'], self.warnings['library'] = create_library(library_type=library_type_obj,
+                                                                                           index=index_obj,
+                                                                                           platform=platform_obj,
+                                                                                           strandedness=library['strandedness'])
+        if not self.errors:
+            sample_obj, self.errors['sample'], self.warnings['sample'] = \
+                create_full_sample(name=sample['name'], volume=sample['volume'], collection_site=sample['collection_site'],
+                                  creation_date=sample['creation_date'], coordinates=sample['coordinates'], alias=sample['alias'],
+                                  concentration=sample['concentration'], tissue_source=sample['tissue_source'],
+                                  experimental_group=sample['experimental_group'], container=container_obj, individual=individual_obj,
+                                  library=library_obj, sample_kind=sample_kind_obj, comment=comment)
```

Comment:
Once this gets executed and the sample created you will have a problem with library_obj. If library is not submitted in the template the library_obj variable will be referenced before assignment. You'd have to assign it before